<!DOCTYPE html>
<html lang="en">

  <!-- Head -->
  <head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">    <!-- Metadata, OpenGraph and Schema.org -->
    

    <!-- Standard metadata -->
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>Publications | Ahmad  Abdel-Azim</title>
    <meta name="author" content="Ahmad  Abdel-Azim">
    <meta name="description" content="Publications and manuscripts in press are included below, categorized by field.">
    <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website">


    <!-- Bootstrap & MDB -->
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous">

    <!-- Bootstrap Table -->
    <link defer rel="stylesheet" href="https://unpkg.com/bootstrap-table@1.21.4/dist/bootstrap-table.min.css">

    <!-- Fonts & Icons -->
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/css/all.min.css" integrity="sha256-mUZM63G8m73Mcidfrv5E+Y61y7a12O5mW4ezU3bxqW4=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous">
    <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons">

    <!-- Code Syntax Highlighting -->
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/github.css" media="" id="highlight_theme_light">

    

    <!-- Styles -->
    
    <link rel="shortcut icon" href="/al-folio/assets/img/webicon.png">
    
    <link rel="stylesheet" href="/al-folio/assets/css/main.css">
    <link rel="canonical" href="https://alshedivat.github.io/al-folio/publications/">

    <!-- Dark Mode -->
    

  </head>

  <!-- Body -->
  <body class="fixed-top-nav sticky-bottom-footer">

    <!-- Header -->
    <header>

      <!-- Nav Bar -->
      <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top">
        <div class="container">
          <a class="navbar-brand title font-weight-lighter" href="/al-folio/"><span class="font-weight-bold">Ahmad Abdel-Azim</span>
          </a>
          <!-- Social Icons -->
          <!-- Navbar Toggle -->
          <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar top-bar"></span>
            <span class="icon-bar middle-bar"></span>
            <span class="icon-bar bottom-bar"></span>
          </button>

          <div class="collapse navbar-collapse text-right" id="navbarNav">
            <ul class="navbar-nav ml-auto flex-nowrap">

              <!-- About -->
              <!-- <li class="nav-item ">
                <a class="nav-link" href="/al-folio/">home</a>
              </li> 
               -->
              
              <!-- Blog -->
              <!-- <li class="nav-item ">
                <a class="nav-link" href="/al-folio/blog/">Blog</a>
              </li> -->

              <!-- Other pages -->
              <li class="nav-item active">
                <a class="nav-link" href="/al-folio/publications/">Publications<span class="sr-only">(current)</span></a>
              </li>
              <li class="nav-item ">
                <a class="nav-link" href="/al-folio/projects/">Projects</a>
              </li>
              <li class="nav-item ">
                <a class="nav-link" href="/al-folio/blog/">Blog</a>
              </li>
              <li class="nav-item ">
                <a class="nav-link" href="/al-folio/cv/">CV</a>
              </li>
            </ul>
          </div>
        </div>
      </nav>

      <!-- Scrolling Progress Bar -->
      <progress id="progress" value="0">
        <div class="progress-container">
          <span class="progress-bar"></span>
        </div>
      </progress>
    </header>


    <!-- Content -->
    <div class="container mt-5">
      
        <!-- page.html -->
        <div class="post">

          <header class="post-header">
            <h1 class="post-title">Publications</h1>
            <p class="post-description">Publications and manuscripts in press are included below, categorized by field.</p>
          </header>

          <article>
            <!-- _pages/publications.md -->
<div class="publications">
  <h2 class="year">Statistical Genetics</h2>
  <div style="height: 50px;"></div>
  <ol class="bibliography"><li>
<!-- _layouts/bib.html -->
      <div class="row">
        <div class="col-sm-2 preview">
<figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/al-folio/assets/img/publication_preview/prs_cover-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/al-folio/assets/img/publication_preview/prs_cover-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/al-folio/assets/img/publication_preview/prs_cover-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/al-folio/assets/img/publication_preview/prs_cover.png" class="preview z-depth-1 rounded" width="auto" height="auto" alt="prs_cover.png" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>
</div>

        <!-- Entry bib key -->
        <div id="prs_paper" class="col-sm-8">
        <!-- Title -->
        <div class="title">Joint estimation of millions of genetic variant effects in Biobank-scale data for accurate early diagnosis and disease progression</div>
        <!-- Author -->
        <div class="author">
        

        <em>A. Abdel-Azim</em>, and X. Lin</div>

        <!-- Journal/Book title and date -->
        
        
        <div class="periodical">
          <em></em> In preparation
        </div>
        <div class="periodical">
          
        </div>

          <!-- Links/Buttons -->
          <div class="links">
            <a class="description btn btn-sm z-depth-0" role="button">Description</a>
          </div>
          
          <div class="badges">
            <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span>
              <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span>
          </div>

          <!-- Hidden description block -->
          <div class="description hidden">
            <p>We are developing novel robust genetic risk score paradigms for patient stratification and early disease diagnosis, leveraging a mixed effects model framework. The method developed is efficient, scalable, and can handle both longitudinal data and related samples, opening the door for genomics-enhanced treatment plans. We are validating the method via simulation and empirically testing the approach using UK Biobank data.</p>
          </div>
        </div>
      </div>
</li></ol>

  <h2 class="year">Cancer Genomics</h2>
  <div style="height: 50px;"></div>
  <ol class="bibliography">
<li>
<!-- _layouts/bib.html -->
      <div class="row">
        <div class="col-sm-2 preview">
<figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/al-folio/assets/img/publication_preview/spectrum_cover-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/al-folio/assets/img/publication_preview/spectrum_cover-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/al-folio/assets/img/publication_preview/spectrum_cover-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/al-folio/assets/img/publication_preview/spectrum_cover.jpeg" class="preview z-depth-1 rounded" width="auto" height="auto" alt="spectrum_cover.jpeg" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>
</div>

        <!-- Entry bib key -->
        <div id="spectrum_tsc_paper" class="col-sm-8">
        <!-- Title -->
        <div class="title">Spectrum of Germline and Somatic Mitochondrial DNA Variants in Tuberous Sclerosis Complex. Frontiers in Genetics</div>
        <!-- Author -->
        <div class="author">
        

        K. Giannikou, K. Martin, <em>A. Abdel-Azim</em>, and
          <span class="more-authors" title="click to view 8 more authors" onclick="
                var element = $(this);
                element.attr('title', '');
                var more_authors_text = element.text() == '8 more authors' ? 'K. Pamir, T. Hougard, S. Bagwe, Y. Tang, J. McKeigan, D. Kwiatkowski, E. Henske, H. Lam' : '8 more authors';
                var cursorPosition = 0;
                var textAdder = setInterval(function(){
                  element.text(more_authors_text.substring(0, cursorPosition + 1));
                  if (++cursorPosition == more_authors_text.length){
                    clearInterval(textAdder);
                  }
              }, '10');
              ">8 more authors</span>
</div>

        <!-- Journal/Book title and date -->
        
        
        <div class="periodical">
          <em>Frontiers in Genetics</em>, Jan 2023
        </div>
        <div class="periodical">
          
        </div>

          <!-- Links/Buttons -->
          <div class="links">
            <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
            <a href="/al-folio/assets/pdf/spectrum_tsc_paper.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
          </div>
          
          <div class="badges">
            <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.3389/fgene.2022.917993"></span>
              <span class="__dimensions_badge_embed__" data-doi="10.3389/fgene.2022.917993" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span>
          </div>

          <!-- Hidden abstract block -->
          <div class="abstract hidden">
            <p>Tuberous Sclerosis Complex (TSC) is caused by loss of function variants in either TSC1 or TSC2 and is characterized by broad phenotypic heterogeneity. Currently, there is limited knowledge regarding the role of the mitochondrial genome (mtDNA) in TSC pathogenesis. In this study, we aimed to determine the prevalence and spectrum of germline and somatic mtDNA variants in TSC and identify potential disease modifiers. Analysis of mtDNA amplicon massively parallel sequencing (aMPS) data, off-target mtDNA from whole-exome sequencing (WES), and/or qPCR, revealed mtDNA alterations in 270 diverse tissues (139 TSC-associated tumors and 131 normal tissue samples) from 199 patients and six healthy individuals. Correlation of clinical features to mtDNA variants and haplogroup analysis was done in 102 buccal swabs (age: 20-71 years). No correlation was found between clinical features and either mtDNA variants or haplogroups. No pathogenic variants were identified in the buccal swab samples. Using in silico analysis, we identified three predicted pathogenic variants in tumor samples: MT-ND4 (m.11742G&gt;A, p. Cys328Tyr, VAF: 43%, kidney angiomyolipoma), MT-CYB (m.14775T&gt;C, p. Leu10Pro, VAF: 43%, LAM abdominal tumor) and MT-CYB (m.15555C&gt;T, p. Pro270Leu, VAF: 7%, renal cell carcinoma). Large deletions of the mitochondrial genome were not detected. Analysis of tumors from 23 patients with corresponding normal tissue did not reveal any recurrent tumor-associated somatic variants. The mtDNA/gDNA ratio between tumors and corresponding normal tissue was also unchanged. Overall, our findings demonstrate that the mitochondrial genome is highly stable across tissues and within TSC-associated tumors.</p>
          </div>
        </div>
      </div>
</li>
<li>
<!-- _layouts/bib.html -->
      <div class="row">
        <div class="col-sm-2 preview">
<figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/al-folio/assets/img/publication_preview/pecoma_cover-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/al-folio/assets/img/publication_preview/pecoma_cover-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/al-folio/assets/img/publication_preview/pecoma_cover-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/al-folio/assets/img/publication_preview/pecoma_cover.png" class="preview z-depth-1 rounded" width="auto" height="auto" alt="pecoma_cover.png" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>
</div>

        <!-- Entry bib key -->
        <div id="pecoma_paper" class="col-sm-8">
        <!-- Title -->
        <div class="title">Comprehensive genomic and clinicopathologic characterization of advanced PEComa tumors with metastatic potential</div>
        <!-- Author -->
        <div class="author">
        

        K. Giannikou, <em>A. Abdel-Azim</em>, J. Tsuji, and
          <span class="more-authors" title="click to view 9 more authors" onclick="
                var element = $(this);
                element.attr('title', '');
                var more_authors_text = element.text() == '9 more authors' ? 'E. Adib, Z. Zhu, C. Wu, E. Martignetti, G. Ha, G. Getz, J. Hornick, A. Wagner, D. Kwiatkowski' : '9 more authors';
                var cursorPosition = 0;
                var textAdder = setInterval(function(){
                  element.text(more_authors_text.substring(0, cursorPosition + 1));
                  if (++cursorPosition == more_authors_text.length){
                    clearInterval(textAdder);
                  }
              }, '10');
              ">9 more authors</span>
</div>

        <!-- Journal/Book title and date -->
        
        
        <div class="periodical">
          <em></em> Jan In preparation
        </div>
        <div class="periodical">
          
        </div>

          <!-- Links/Buttons -->
          <div class="links">
            <a class="description btn btn-sm z-depth-0" role="button">Description</a>
          </div>
          
          <div class="badges">
            <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span>
              <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span>
          </div>

          <!-- Hidden description block -->
          <div class="description hidden">
            <p>Background: Malignant PEComa (Perivascular Epithelioid cell tumor) is a rare
  neoplasm that is predominantly seen in females. This tumor type affects multiple organs and
  demonstrates local invasion and/or distant metastases with poor prognosis. Approximately 35-40%
  of PEComa have mutations in TSC1/TSC2 genes. Aim: To identify other driver events beyond
  TSC1/TSC2 genetic alterations that might contribute to tumor development and to compare to The
  Cancer Genome Atlas (TCGA) Leiomyosarcoma (LMS), a similar sarcoma subtype with overlapping 
  morphologic and immunophenotypic features which can make diagnosis challenging. Material-
  Methods: Exome sequencing was performed in 74 malignant PEComa tumors from 59 patients
  (median age 52). This is the first exome analysis conducted in the largest cohort of malignant
  PEComa tumors ever assembled. Results: We found that mutations in TP53 (27%), TSC2 (24%) and
  TSC1 (18%) were the most frequent alterations, whereas other somatic events were mainly non-
  recurrent missense alterations of uncertain clinical significance. C&gt;T at CpG was the most common
  substitution, consistent with the aging mutation signature, similar to LMS. Copy number analysis
  showed that PEComa have multiple focal amplifications and deletions (e.g., RB1, TP53), also similar
  to LMS. Many of these PEComa patients (75%) had been treated with rapamycin or other mTORC1
  inhibitors, as mutations in TSC1/TSC2 in tumors are associated with response to treatment with
  mTORC1 inhibitors (mTORi). Patients with TSC2 mutations had sustained clinical benefit in
  response to mTORi, median progression-free survival 1.4 years, and median overall survival 2.9
  years. Conclusions: PEComa have a low somatic mutation rate, similar to LMS, and substantial
  TSC1/TSC2 mutations which are not seen in LMS. PEComa patients respond to rapalog therapy with
  favorable disease outcome and such responses are often durable (&gt;1 year) and strongly correlate
  with alterations in TSC2.</p>
          </div>
        </div>
      </div>
</li>
<li>
<!-- _layouts/bib.html -->
      <div class="row">
        <div class="col-sm-2 preview">
<figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/al-folio/assets/img/publication_preview/peclms_cover-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/al-folio/assets/img/publication_preview/peclms_cover-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/al-folio/assets/img/publication_preview/peclms_cover-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/al-folio/assets/img/publication_preview/peclms_cover.png" class="preview z-depth-1 rounded" width="auto" height="auto" alt="peclms_cover.png" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>
</div>

        <!-- Entry bib key -->
        <div id="peclms_paper" class="col-sm-8">
        <!-- Title -->
        <div class="title">Distinct oncogenic signatures in malignant PEComa and leiomyosarcoma identified by integrative RNA-seq and H3K27ac ChIP-seq analysis</div>
        <!-- Author -->
        <div class="author">
        

        K. Giannikou, <em>A. Abdel-Azim</em>, E. Adib, and
          <span class="more-authors" title="click to view 10 more authors" onclick="
                var element = $(this);
                element.attr('title', '');
                var more_authors_text = element.text() == '10 more authors' ? 'I. Schaefer, N. Kesten, L. Taing, Z. Zhu, J. Hornick, M. Hirsch, H. Long, A. Wagner, M. Hemming, D. Kwiatkowski' : '10 more authors';
                var cursorPosition = 0;
                var textAdder = setInterval(function(){
                  element.text(more_authors_text.substring(0, cursorPosition + 1));
                  if (++cursorPosition == more_authors_text.length){
                    clearInterval(textAdder);
                  }
              }, '10');
              ">10 more authors</span>
</div>

        <!-- Journal/Book title and date -->
        
        
        <div class="periodical">
          <em></em> Jan In preparation
        </div>
        <div class="periodical">
          
        </div>

          <!-- Links/Buttons -->
          <div class="links">
            <a class="description btn btn-sm z-depth-0" role="button">Description</a>
          </div>
          
          <div class="badges">
            <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span>
              <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span>
          </div>

          <!-- Hidden description block -->
          <div class="description hidden">
            <p>Malignant Perivascular epithelioid cell tumor (PEComa) and leiomyosarcoma (LMS) are two
  sarcomas with overlapping morphologic and immunophenotypic features that can make them
  difficult to distinguish, leading to diagnostic uncertainty. We performed integrative RNA-seq and
  H3K27ac ChIP-seq analysis of PEComa and LMS and observed that PEComa and LMS co-
  cluster in comparison to six other sarcoma subtypes. Multiple differentially expressed and
  H3K27ac-marked genes were identified between groups, including DAPL1, MLANA, SULT1C2,
  GPR143, and CHI3L1 higher in PEComa; and MYOCD, WFDC2, DES, MYH11, and CNN1
  higher in LMS. Gene set enrichment analysis showed higher expression of lysosome genes in
  PEComa; and myogenesis and smooth muscle contraction genes in LMS. Differentially
  expressed genes between these two sarcomas were assessed as candidate biomarkers by
  immunohistochemistry (IHC). PNL2, PMEL, and MLANA all demonstrated complete specificity
  for PEComa over LMS, but relatively low sensitivity, desmin positivity by IHC was enriched in
  LMS, while other select IHC markers could not discriminate between these two in an
  independent cohort. Given the success of mTORC1 inhibitors in the treatment of TSC1/TSC2-
  mutant PEComa and the potential misdiagnosis between LMS and PEComa, genetic analysis
  for TSC1/TSC2 mutations should be considered for all ambiguous LMS patients.</p>
          </div>
        </div>
      </div>
</li>
</ol>

  <h2 class="year">Microbiology</h2>
  <div style="height: 50px;"></div>
  <ol class="bibliography">
<li>
<!-- _layouts/bib.html -->
      <div class="row">
        <div class="col-sm-2 preview">
<figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/al-folio/assets/img/publication_preview/phcog_cover-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/al-folio/assets/img/publication_preview/phcog_cover-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/al-folio/assets/img/publication_preview/phcog_cover-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/al-folio/assets/img/publication_preview/phcog_cover.png" class="preview z-depth-1 rounded" width="auto" height="auto" alt="phcog_cover.png" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>
</div>

        <!-- Entry bib key -->
        <div id="phcog_paper" class="col-sm-8">
        <!-- Title -->
        <div class="title">Characterization of the Gain and Loss of Resistance to Antibiotics versus Tolerance to Honey as an Antimutagenic and Antimicrobial Medium in Extended‑Time Serial Transfer Experiments</div>
        <!-- Author -->
        <div class="author">
        

        <em>A. Abdel-Azim</em>, S. Abdel-Azim, B. Piasecki, and
          <span class="more-authors" title="click to view 1 more author" onclick="
                var element = $(this);
                element.attr('title', '');
                var more_authors_text = element.text() == '1 more author' ? 'G. Abdel-Azim' : '1 more author';
                var cursorPosition = 0;
                var textAdder = setInterval(function(){
                  element.text(more_authors_text.substring(0, cursorPosition + 1));
                  if (++cursorPosition == more_authors_text.length){
                    clearInterval(textAdder);
                  }
              }, '10');
              ">1 more author</span>
</div>

        <!-- Journal/Book title and date -->
        
        
        <div class="periodical">
          <em>Pharmacognosy Research</em>, Apr 2019
        </div>
        <div class="periodical">
          
        </div>

          <!-- Links/Buttons -->
          <div class="links">
            <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
            <a href="/al-folio/assets/pdf/phcog_paper.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
          </div>
          
          <div class="badges">
            <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right" data-doi="10.4103/pr.pr_175_18"></span>
              <span class="__dimensions_badge_embed__" data-doi="10.4103/pr.pr_175_18" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span>
          </div>

          <!-- Hidden abstract block -->
          <div class="abstract hidden">
            <p>Background: Honey contains several substances with antimicrobial properties that appear more recalcitrant to generating bacterial‑evolved resistance than traditional antibiotics. Objectives: This study seeks to characterize the evolution of bacteria grown for successive generations in honey as an antimutagenic medium versus ampicillin. Materials and Methods: A naive strain of Staphylococcus aureus was serially cultured for 17 days in Lysogeny broth (LB) containing sublethal concentrations of medical‑grade Manuka honey, polyfloral honey, and ampicillin. Glucose as an osmotic control and pure LB were included for comparison. A portion of each culture was removed every 24 h to (a) determine the amount of growth that occurred during the previous 24 h and (b) use as a genetic stock in serially transferred tubes containing the same inhibitory compound as used previously. Results: As indicated by an increase in growth over sequential 24‑h period, bacteria rapidly gained resistance to ampicillin in a step‑wise pattern. However, bacteria grown in Manuka and polyfloral honey never exceeded their initial growth levels. Bacteria grown in relatively high concentrations of honey for a single 24‑h period consistently lost viability after one transfer, a phenomenon that has not been reported in the literature before and which indicates the inability of bacteria to adapt to honey as an antimutagenic medium. While bacteria grown in honey did not evolve the ability to grow at higher concentrations, a single isolate grown in Manuka honey gained tolerance to Manuka by successfully surviving and proliferating beyond second transfers. Bacteria that developed antibiotic resistance were found to remain sensitive to honey. Moreover, bacteria lost their resistance to ampicillin upon a single exposure to Manuka honey.</p>
          </div>
        </div>
      </div>
</li>
<li>
<!-- _layouts/bib.html -->
      <div class="row">
        <div class="col-sm-2 preview">
<figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/al-folio/assets/img/publication_preview/osmotic_cover-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/al-folio/assets/img/publication_preview/osmotic_cover-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/al-folio/assets/img/publication_preview/osmotic_cover-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/al-folio/assets/img/publication_preview/osmotic_cover.png" class="preview z-depth-1 rounded" width="auto" height="auto" alt="osmotic_cover.png" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>
</div>

        <!-- Entry bib key -->
        <div id="osmotic_paper" class="col-sm-8">
        <!-- Title -->
        <div class="title">Determining the Contribution of Osmotic Stress to the Antibacterial Properties of Honey</div>
        <!-- Author -->
        <div class="author">
        

        <em>A. Abdel-Azim</em>, S. Abdel-Azim, and G. Abdel-Azim</div>

        <!-- Journal/Book title and date -->
        
        
        <div class="periodical">
          <em>Journal of Emerging Investigators</em>, Mar 2019
        </div>
        <div class="periodical">
          
        </div>

          <!-- Links/Buttons -->
          <div class="links">
            <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
            <a href="/al-folio/assets/pdf/osmotic_paper.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
          </div>
          
          <div class="badges">
            <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span>
              <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span>
          </div>

          <!-- Hidden abstract block -->
          <div class="abstract hidden">
            <p>Researchers have repeatedly shown that honey possesses distinctive antimicrobial properties; however, there is uncertainty over which compounds in honey are responsible for these properties. In this research study, we sought to quantify the role of osmotic stress in honey and determine the efficacy of two types of honey: Manuka and raw pasture honey. Bacteria were sequentially cultured in sublethal concentrations of Manuka and raw pasture honey for five days. The role of osmotic stress as a contributor to the antibacterial properties of Manuka and raw pasture honey was quantified in the first culture and over five serial cultures. The growth levels of bacteria in honey were compared to growth levels in glucose, an osmotic control, to quantify the role of osmotic stress in the two types of honey. The results of this study indicate that in the first culture, the antibacterial impacts of Manuka and raw pasture honey were primarily attributable to osmotic stress. However, over five days of sequential transfers, both raw pasture honey and Manuka honey showed significant antibacterial properties beyond osmotic strength. It was established that the antibacterial properties of honey cannot be investigated based solely on the first culture. Serial transfers over several days should be employed to investigate the efficacy of honey as an antibacterial substance.</p>
          </div>
        </div>
      </div>
</li>
<li>
<!-- _layouts/bib.html -->
      <div class="row">
        <div class="col-sm-2 preview">
<figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/al-folio/assets/img/publication_preview/resistance_cover-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/al-folio/assets/img/publication_preview/resistance_cover-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/al-folio/assets/img/publication_preview/resistance_cover-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/al-folio/assets/img/publication_preview/resistance_cover.jpg" class="preview z-depth-1 rounded" width="auto" height="auto" alt="resistance_cover.jpg" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>
</div>

        <!-- Entry bib key -->
        <div id="resistance_paper" class="col-sm-8">
        <!-- Title -->
        <div class="title">The Impact of Antibiotic Exposure and Concentration on Resistance in Bacteria</div>
        <!-- Author -->
        <div class="author">
        

        <em>A. Abdel-Azim</em>, and G. Abdel-Azim</div>

        <!-- Journal/Book title and date -->
        
        
        <div class="periodical">
          <em>Journal of Emerging Investigators</em>, Oct 2016
        </div>
        <div class="periodical">
          
        </div>

          <!-- Links/Buttons -->
          <div class="links">
            <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
            <a href="/al-folio/assets/pdf/resistance_paper.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
          </div>
          
          <div class="badges">
            <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span>
              <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span>
          </div>

          <!-- Hidden abstract block -->
          <div class="abstract hidden">
            <p>How many generations are needed for bacteria to gain antibiotic resistance and to what extent does antibiotic concentration make a difference? To address these questions, pure Escherichia coli was cultured on nutrient agar in the presence of low, medium, and high concentrations of amoxicillin. Surviving bacteria were repeatedly cultured for four consecutive generations, and the sensitivity of bacteria was measured within each generation. We measured antibiotic sensitivity by evaluating the diameter of the inhibition zone around antibiotic-saturated discs of all three amoxicillin concentrations. Although exposing bacteria to extremely heavy antibiotic concentrations resulted in greater inhibition zones around antibiotic-saturated discs, there were no major differences in sensitivity relative to the much lower concentrations studied. We also found that bacteria gained significant resistance upon first exposure to the antibiotic. In fact, the reduction in average inhibition zones was much greater in early generations than in late generations. We conclude that E. coli can gain major resistance upon surviving early exposure to amoxicillin, which suggests that antibiotic courses of medication should be prescribed to completely clear the body of bacterial infection upon first antibiotic exposure.</p>
          </div>
        </div>
      </div>
</li>
<li>
<!-- _layouts/bib.html -->
      <div class="row">
        <div class="col-sm-2 preview">
<figure>

  <picture>
    
    <source class="responsive-img-srcset" media="(max-width: 480px)" srcset="/al-folio/assets/img/publication_preview/QS_cover-480.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 800px)" srcset="/al-folio/assets/img/publication_preview/QS_cover-800.webp"></source>
    <source class="responsive-img-srcset" media="(max-width: 1400px)" srcset="/al-folio/assets/img/publication_preview/QS_cover-1400.webp"></source>
    

    <!-- Fallback to the original file -->
    <img src="/al-folio/assets/img/publication_preview/QS_cover.png" class="preview z-depth-1 rounded" width="auto" height="auto" alt="QS_cover.png" onerror="this.onerror=null; $('.responsive-img-srcset').remove();">

  </picture>

</figure>
</div>

        <!-- Entry bib key -->
        <div id="QS_paper" class="col-sm-8">
        <!-- Title -->
        <div class="title">The characterization of quorum sensing trajectories of Vibrio fischeri using longitudinal data analytics</div>
        <!-- Author -->
        <div class="author">
        

        M. Abdel-Azim, and <em>A. Abdel-Azim</em>
</div>

        <!-- Journal/Book title and date -->
        
        
        <div class="periodical">
          <em>Journal of Emerging Investigators</em>, Dec 2023
        </div>
        <div class="periodical">
          
        </div>

          <!-- Links/Buttons -->
          <div class="links">
            <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
            <a href="/al-folio/assets/pdf/QS_paper.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
          </div>
          
          <div class="badges">
            <span class="altmetric-embed" data-hide-no-mentions="true" data-hide-less-than="15" data-badge-type="2" data-badge-popover="right"></span>
              <span class="__dimensions_badge_embed__" data-pmid="" data-hide-zero-citations="true" data-style="small_rectangle" data-legend="hover-right" style="margin-bottom: 3px;"></span>
          </div>

          <!-- Hidden abstract block -->
          <div class="abstract hidden">
            <p>Quorum sensing (QS) is the process in which bacteria recognize and respond to the surrounding cell density, and it can be inhibited by certain antimicrobial substances. In Vibrio fischeri, bioluminescence intensifies as more cells utilize QS, making it an accurate biomarker for QS. In this study, we showed that observed illumination intensity data measured at multiple time points is insufficient for evaluating and comparing QS activity across different cultures without proper statistical modeling. We introduced a statistical approach to characterize illumination trajectories over time using longitudinal data analytics. The approach was tested on the naturally illuminating bacteria, V. fischeri, which emit light when QS is activated, but it can also be implemented with green fluorescent protein (GFP) and its numerous applications. To compare multiple QS trajectories, we suppressed QS activity at varying levels using honey, known for its antimicrobial and anti-QS properties. We characterized and compared illumination trajectories through time in bacterial cultures containing five Manuka honey concentrations. The longitudinal approach analyzed illumination data throughout the entire QS cycle and correctly ranked bacterial cultures according to their true level of QS activity. Conversely, the conventional cross-sectional approach was inconsistent in evaluating bacteria for their QS activity and trajectory. We conclude that modeling illumination intensity through time provides a more accurate evaluation of QS activity than conventional cross-sectional analysis.</p>
          </div>
        </div>
      </div>
</li>
</ol>


</div>

          </article>

        </div>

      
    </div>

    <!-- Footer -->    <!-- <footer class="sticky-bottom mt-5">
      <div class="container">
        &copy; Copyright 2023 Ahmad  Abdel-Azim. Powered by <a href="https://jekyllrb.com/" target="_blank">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank">GitHub Pages</a>. Photos from <a href="https://unsplash.com" target="_blank">Unsplash</a>.

      </div>
    </footer> -->

    <!-- JavaScripts -->
    <!-- jQuery -->
  <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>

    <!-- Bootsrap & MDB scripts -->
  <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js" integrity="sha256-fgLAgv7fyCGopR/gBNq2iW3ZKIdqIcyshnUULC4vex8=" crossorigin="anonymous"></script>
  <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script>

    <!-- Masonry & imagesLoaded -->
  <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script>
  <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script>
  <script defer src="/al-folio/assets/js/masonry.js" type="text/javascript"></script>
    
  <!-- Medium Zoom JS -->
  <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.8/dist/medium-zoom.min.js" integrity="sha256-7PhEpEWEW0XXQ0k6kQrPKwuoIomz8R8IYyuU1Qew4P8=" crossorigin="anonymous"></script>
  <script defer src="/al-folio/assets/js/zoom.js"></script>

  <!-- Bootstrap Table -->
  <script defer src="https://unpkg.com/bootstrap-table@1.21.4/dist/bootstrap-table.min.js"></script>

  <!-- Load Common JS -->
  <script src="/al-folio/assets/js/no_defer.js"></script>
  <script defer src="/al-folio/assets/js/common.js"></script>
  <script defer src="/al-folio/assets/js/copy_code.js" type="text/javascript"></script>

    
  <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script>
  <script async src="https://badge.dimensions.ai/badge.js"></script>

    <!-- MathJax -->
  <script type="text/javascript">
    window.MathJax = {
      tex: {
        tags: 'ams'
      }
    };
  </script>
  <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script>
  <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>

    
    

<!-- Scrolling Progress Bar -->
<script type="text/javascript">
  /*
   * This JavaScript code has been adapted from the article 
   * https://css-tricks.com/reading-position-indicator/ authored by Pankaj Parashar, 
   * published on the website https://css-tricks.com on the 7th of May, 2014.
   * Couple of changes were made to the original code to make it compatible 
   * with the `al-foio` theme.
   */
  const progressBar = $("#progress");
  /*
   * We set up the bar after all elements are done loading.
   * In some cases, if the images in the page are larger than the intended
   * size they'll have on the page, they'll be resized via CSS to accomodate
   * the desired size. This mistake, however, breaks the computations as the
   * scroll size is computed as soon as the elements finish loading.
   * To account for this, a minimal delay was introduced before computing the
   * values.
   */
  window.onload = function () {
    setTimeout(progressBarSetup, 50);
  };
  /*
   * We set up the bar according to the browser.
   * If the browser supports the progress element we use that.
   * Otherwise, we resize the bar thru CSS styling
   */
  function progressBarSetup() {
    if ("max" in document.createElement("progress")) {
      initializeProgressElement();
      $(document).on("scroll", function() {
        progressBar.attr({ value: getCurrentScrollPosition() });
      });
      $(window).on("resize", initializeProgressElement);
    } else {
      resizeProgressBar();
      $(document).on("scroll", resizeProgressBar);
      $(window).on("resize", resizeProgressBar);
    }
  }
  /*
   * The vertical scroll position is the same as the number of pixels that
   * are hidden from view above the scrollable area. Thus, a value > 0 is
   * how much the user has scrolled from the top
   */
  function getCurrentScrollPosition() {
    return $(window).scrollTop();
  }

  function initializeProgressElement() {
    let navbarHeight = $("#navbar").outerHeight(true);
    $("body").css({ "padding-top": navbarHeight });
    $("progress-container").css({ "padding-top": navbarHeight });
    progressBar.css({ top: navbarHeight });
    progressBar.attr({
      max: getDistanceToScroll(),
      value: getCurrentScrollPosition(),
    });
  }
  /*
   * The offset between the html document height and the browser viewport
   * height will be greater than zero if vertical scroll is possible.
   * This is the distance the user can scroll
   */
  function getDistanceToScroll() {
    return $(document).height() - $(window).height();
  }

  function resizeProgressBar() {
    progressBar.css({ width: getWidthPercentage() + "%" });
  }
  // The scroll ratio equals the percentage to resize the bar
  function getWidthPercentage() {
    return (getCurrentScrollPosition() / getDistanceToScroll()) * 100;
  }
</script>

  </body>
</html>
